adapters/iqzone/iqzone.go

```diff
@@ -0,0 +1,236 @@
+package iqzone
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type IQZoneAdapter struct {
+	EndpointTemplate template.Template
+}
+
+//MakeRequests prepares request information for prebid-server core
+func (adapter *IQZoneAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+	if len(request.Imp) == 0 {
+		errs = append(errs, &errortypes.BadInput{Message: "No impression in the bid request"})
+		return nil, errs
+	}
+	pub2impressions, imps, err := getImpressionsInfo(request.Imp)
+	if len(imps) == 0 {
+		return nil, err
+	}
+	errs = append(errs, err...)
+
+	if len(pub2impressions) == 0 {
+		return nil, errs
+	}
+
+	result := make([]*adapters.RequestData, 0, len(pub2impressions))
+	for k, imps := range pub2impressions {
+		bidRequest, err := adapter.buildAdapterRequest(request, &k, imps)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
```

Comment:
Should this be a continue instead to allow for a partial request of the valid imps, or do you want to specifically error out if one of the imps is invalid? I wouldn't normally expect errors from the `buildAdapterRequest`. If it does error, something is likely VERY wrong.. along the lines of memory corruption since the errors are only generated by marshal attemps and template building.

Reply:
Thanks, i added to continue with valid impression.